### PR TITLE
Add ability to disable audit logs

### PR DESCRIPTION
**Pull request by wickloww**
## Description

Add ability to disable audit logs in admin panel

## Related issues

Fixes https://github.com/Chocobozzz/PeerTube/issues/2686

## Has this been tested?

- [x] 👍 yes, I added tests to the test suite
- [ ] 💭 no, because this PR is a draft and still needs work
- [ ] 🙅 no, because this PR does not update server code
- [ ] 🙋 no, because I need help 

## Screenshots

image

---
image

### CODE REVIEW

File: config/default.yaml [hunk 1]
```diff
@@ -763,6 +763,9 @@ instance:
   securitytxt: |
     Contact: https://github.com/Chocobozzz/PeerTube/blob/develop/SECURITY.md
     Expires: 2025-12-31T11:00:00.000Z'
+  logs:
```
Comment (Chocobozzz) [inline config/default.yaml:766]: You already have a log section: https://github.com/Chocobozzz/PeerTube/blob/develop/config/default.yaml#L230

Please also add this block to `production.yaml.example`

File: client/src/app/+admin/config/edit-custom-config/edit-advanced-configuration.component.html [hunk 2]
```diff
@@ -136,4 +136,27 @@ <h2 i18n class="inner-form-title">CUSTOMIZATIONS</h2>
     </div>
   </div>
 
+  <div class="row mt-4"> <!-- cache grid -->
```
Comment (Chocobozzz) [inline client/src/app/+admin/config/edit-custom-config/edit-advanced-configuration.component.html:139]: You can remove HTML comments

File: client/src/app/+admin/config/edit-custom-config/edit-advanced-configuration.component.html [hunk 3]
```diff
@@ -136,4 +136,27 @@ <h2 i18n class="inner-form-title">CUSTOMIZATIONS</h2>
...
+    <div class="col-12 col-lg-4 col-xl-3">
+      <div class="anchor" id="customizations"></div> <!-- customizations anchor -->
+      <h2 i18n class="inner-form-title">LOGS</h2>
+      <div i18n class="inner-form-description">
```
Comment (Chocobozzz) [inline client/src/app/+admin/config/edit-custom-config/edit-advanced-configuration.component.html:143]: The description of this block is invalid

File: client/src/app/+admin/system/logs/logs.component.html [hunk 4]
```diff
@@ -30,14 +31,16 @@
...
-  <my-button i18n-label label="Refresh" icon="refresh" (click)="refresh()"></my-button>
+  <my-button *ngIf="!isAuditLog() || isAuditLogsEnabled" i18n-label label="Refresh" icon="refresh" (click)="refresh()"></my-button>
 </div>
 
 <div class="logs">
   <div *ngIf="loading" i18n>Loading...</div>
 
   <div #logsElement>
-    <div *ngIf="!loading && logs.length === 0" i18n>No log.</div>
+    <div *ngIf="isAuditLog && !isAuditLogsEnabled" i18n>Audit logs disabled.</div>
```
Comment (Chocobozzz) [inline client/src/app/+admin/system/logs/logs.component.html:41]: `isAuditLog` is a function so `isAuditLog()`

Also transform `isAuditLogsEnabled` to a function. `is..` is for functions, if you still want to have a variable then prefer `auditLogsEnabled`. But it's better to have a function to not mix variables/functions if we can avoid

File: client/src/app/+admin/system/logs/logs.component.html [hunk 5]
```diff
@@ -30,14 +31,16 @@
...
+
+    <div *ngIf="!loading && logs.length === 0 && isAuditLogsEnabled" i18n>No log.</div>
```
Comment (Chocobozzz) [inline client/src/app/+admin/system/logs/logs.component.html:43]: If audit logs is disabled but you're on the normal logs, you'll never see this message

I suggest to simplify lines 41/43 using an Angular else for example (not easy to use, but will be simpler in the future with the new Angular markup language: https://blog.angular.io/meet-angulars-new-control-flow-a02c6eee7843).

File: packages/tests/src/api/server/logs.ts [hunk 6]
```diff
@@ -193,6 +193,41 @@ describe('Test logs', function () {
...
+              auditLogs:{
+                enabled: false
+              }
+            }
+          }
+        }
+      })
+
+      await server.videos.upload({ attributes: { name: 'video 12' } })
+      await waitJobs([ server ])
```
Comment (Chocobozzz) [inline packages/tests/src/api/server/logs.ts:213]: Just upload one video to test audit logs. Test 403 if disabled, but also the audit log was not filled when it was disabled (so re-enable the feature and check the audit log doesn't contain your upload)

**Review by Chocobozzz — CHANGES_REQUESTED** [review]

**Comment by ThisIsMissEm** [comment]
You may not want to actually have this as a feature, as it may violate some of the requirements for the Digital Services Act (e.g., to be able to provide evidence as to what an admin/moderator did and when.

The only case that I could see this being argued for is on a single-user instance, but even then audit logs are useful in case you forget doing something.

**Comment by Chocobozzz** [comment]
Audit logs would be enabled by default. But the instance is owned by the admin, so we give them the choice as this feature has been requested by some of them :)

**Comment by ThisIsMissEm** [comment]
> Audit logs would be enabled by default. But the instance is owned by the admin, so we give them the choice as this feature has been requested by some of them :)

Maybe so, but they're still an important feature for trust & safety (and as noted in other issues can be used to implement features for things like actions taken on reports)

It'd be good to know *why* they want to disable the audit logs? What problem are the audit logs creating that could necessitate them needing to be disabled?

**Comment by Chocobozzz** [comment]
Closing due to inactivity. Does not hesitate to ask for the reopening :)

**Pull request closed without merging**